### ISSUE TITLE
Feature Request: Make use of transformers + LLM-AWQ / AutoAWQ integration 🚀 

### ISSUE THREAD

**Issue by younesbelkada** [oobabooga/text-generation-webui#4542]
Hi @oobabooga 

First of all thanks a lot for this great project, and very glad that it uses many tools from HF ecosystem such as quantization! 
Recently we shipped AWQ integration in transformers (since 4.35.0): https://huggingface.co/docs/transformers/main_classes/quantization#awq-integration

TLDR is that you can run models from TheBloke repositories out of the box using `from_pretrained` with no extra argument ! 

I am also exploring if using fused modules could help speeding up generation: https://github.com/huggingface/transformers/pull/27411 

I would like to see if you are interested to have AWQ + transformers in text-generation-webui. I would love to help here and try my hands on integrating this feature if you provide me some guidance. 

Thanks! 

cc @casper-hansen @TheBloke

**Comment by casper-hansen** [comment]
Perhaps the LLM-AWQ part could be interesting to integrate @younesbelkada since AutoAWQ is already integrated

**Comment by oobabooga** [comment]
@younesbelkada I am not familiar with LLM-AWQ, but if it provides some benefit over the current AutoAWQ integration, a PR would be welcome and prioritized.

The transformers version used in the project is already `transformers==4.35.*`, so I assume loading an AWQ model through the transformers loader should already work, like it does for GPTQ models. If any extra optional parameters are available, they can be added as new command-line arguments in `modules/shared.py` and a new matching checkbox in `modules/ui_model_menu.py`. 

The easiest way to add a parameter is to grep for an existing parameter, say `load_in_4bit`, and add the new parameter in all places where the existing one appears.

**Comment by younesbelkada** [comment]
Hi @oobabooga 
Sorry for my late reply and thanks for your response - I will think of something that can be done and raise a PR, one thing that could be interesting is to integrate fused modules https://github.com/huggingface/transformers/pull/27411 but I think this is already in place currently in text-generation-webui right @casper-hansen ?

**Comment by casper-hansen** [comment]
> Hi @oobabooga Sorry for my late reply and thanks for your response - I will think of something that can be done and raise a PR, one thing that could be interesting is to integrate fused modules [huggingface/transformers#27411](https://github.com/huggingface/transformers/pull/27411) but I think this is already in place currently in text-generation-webui right @casper-hansen ?

Yeah TGW already integrates with AutoAWQ and implements fused modules. For llm-awq, it may just be that the normal transformer loader will work since it's integrated in transformers?
